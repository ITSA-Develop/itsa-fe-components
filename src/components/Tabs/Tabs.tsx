import { Tabs as AntTabs, TabsProps } from 'antd';

export const Tabs = ({ ...rest }: TabsProps) => {
	return <AntTabs {...rest} />;
};
