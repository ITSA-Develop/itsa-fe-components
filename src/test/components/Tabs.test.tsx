import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it } from 'vitest';
import { Tabs } from '../../components/Tabs/Tabs';

const items = [
	{ key: '1', label: 'Tab 1', children: <div>Content 1</div> },
	{ key: '2', label: 'Tab 2', children: <div>Content 2</div> },
];

describe('Tabs component', () => {
	it('renders tabs and switches content', async () => {
		const { container } = render(<Tabs defaultActiveKey="1" items={items} />);

		expect(screen.getByText('Content 1')).toBeInTheDocument();
		expect(screen.queryByText('Content 2')).not.toBeInTheDocument();

		await userEvent.click(screen.getByText('Tab 2'));
		expect(screen.getByText('Content 2')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
