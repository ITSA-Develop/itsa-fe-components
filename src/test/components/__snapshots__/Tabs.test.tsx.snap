// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs component > renders tabs and switches content 1`] = `
<div>
  <div
    class="ant-tabs ant-tabs-top css-dev-only-do-not-override-l9pxc0"
  >
    <div
      aria-orientation="horizontal"
      class="ant-tabs-nav"
      role="tablist"
    >
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-list"
          style="transform: translate(0px, 0px); transition: none;"
        >
          <div
            class="ant-tabs-tab"
            data-node-key="1"
          >
            <div
              aria-controls="rc-tabs-test-panel-1"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-1"
              role="tab"
              tabindex="-1"
            >
              Tab 1
            </div>
          </div>
          <div
            class="ant-tabs-tab ant-tabs-tab-active"
            data-node-key="2"
          >
            <div
              aria-controls="rc-tabs-test-panel-2"
              aria-selected="true"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-2"
              role="tab"
              tabindex="0"
            >
              Tab 2
            </div>
          </div>
          <div
            class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            style="width: 0px; left: 0px; transform: translateX(-50%);"
          />
        </div>
      </div>
      <div
        class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      >
        <button
          aria-controls="rc-tabs-test-more-popup"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="ant-tabs-nav-more"
          id="rc-tabs-test-more"
          style="visibility: hidden; order: 1;"
          type="button"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-tabs-content-holder"
    >
      <div
        class="ant-tabs-content ant-tabs-content-top"
      >
        <div
          aria-hidden="true"
          aria-labelledby="rc-tabs-test-tab-1"
          class="ant-tabs-tabpane ant-tabs-tabpane-hidden"
          id="rc-tabs-test-panel-1"
          role="tabpanel"
          tabindex="-1"
        >
          <div>
            Content 1
          </div>
        </div>
        <div
          aria-hidden="false"
          aria-labelledby="rc-tabs-test-tab-2"
          class="ant-tabs-tabpane ant-tabs-tabpane-active"
          id="rc-tabs-test-panel-2"
          role="tabpanel"
          tabindex="0"
        >
          <div>
            Content 2
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
